app/src/main/java/com/devindi/wallpaper/model/map/TileUtils.kt

```diff
@@ -0,0 +1,54 @@
+package com.devindi.wallpaper.model.map
+
+import com.devindi.wallpaper.misc.RectD
+import org.osmdroid.tileprovider.cachemanager.CacheManager
+import timber.log.Timber
+
+class TileUtils {
+
+    fun getTilesRect(
+        centerLat: Double,
+        centerLon: Double,
+        width: Int,
+        height: Int,
+        tileSize: Int,
+        zoom: Int
+    ): RectD {
+        val centerTileY = latToTile(centerLat, zoom)
+        val centerTileX = lonToTile(centerLon, zoom)
+        val tileWidth = width / (2.0 * tileSize)
+        var westTileX = centerTileX - tileWidth
+        val eastTileX = centerTileX + tileWidth
+        val tileHeight = height / (2.0 * tileSize)
+        val northTileY = centerTileY - tileHeight
+        val southTileY = centerTileY + tileHeight
+
+        if (westTileX < 0) {
+            val worldSize = Math.pow(2.0, zoom.toDouble()).toInt()
+            westTileX += worldSize
+        }
+        Timber.d("Tiles rect: $westTileX - $eastTileX; $$northTileY - $southTileY")
+
+        return RectD(westTileX, northTileY, eastTileX, southTileY)
+    }
+
+    // Here is some math magic). Those calculations are taken from OSM wiki
+    // https://wiki.openstreetmap.org/wiki/Slippy_map_tilenames
+    fun lonToTile(lon: Double, zoom: Int) = (lon + HALF_CIRCLE_DEGREES) / CIRCLE_DEGREES * (1 shl zoom)
+
+    fun latToTile(lat: Double, zoom: Int): Double = (1 shl zoom) * (1 - (Math.log(Math.tan(Math.toRadians(lat)) + 1 / (Math.cos(Math.toRadians(lat))))) / Math.PI) / 2
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/devindi/mapwall/blob/728b2e488be620e35d2325d61c174b7d35c33030/app/src/main/java/com/devindi/wallpaper/model/map/TileUtils.kt#L39"></span>Exceeded max line length (100) (cannot be auto-corrected)</td>
    </tr>
  </tbody>
</table>